### DISCUSSION TITLE
Multiple TP strategy with SL

### DISCUSSION THREAD

**Discussion by ehsan-panahi-ce** [kernc/backtesting.py#638] [General]
Hi,
I want to implement a strategy that have multiple tp. when the price hit TP1 (TP1=1.500) , instead of closing the trade, I want to update the TP to TP2 (TP2 =1.600). and so on.

**Comment by ehsan-panahi-ce** [comment]
Hi any news?

**Comment by kernc** [comment]
When a price hits TP (set with `tp=`), the trade is closed. There's no way around it.

If instead you want to set TP to "TP2" once the price hits a certain level, you can set [`Trade.tp`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Trade.tp) property, e.g.:
```py
if self.data.Close[-1] >= 1.5:
    for trade in self.trades:
        trade.tp = 1.6
```
